src/lvm/volume_provisioning/volume_provisioning_test.go

```diff
@@ -0,0 +1,135 @@
+package volume_provisioning
+
+import (
+	"testing"
+	"time"
+
+	"github.com/openebs/openebs-e2e/common/e2e_agent"
+	"github.com/openebs/openebs-e2e/common/e2e_ginkgo"
+	"github.com/openebs/openebs-e2e/common/lvm"
+
+	"github.com/openebs/openebs-e2e/common"
+	"github.com/openebs/openebs-e2e/common/k8stest"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+var nodeConfig lvm.LvmNodesDevicePvVgConfig
+
+func volumeProvisioningTest(decor string, engine common.OpenEbsEngine, volType common.VolumeType, fstype common.FileSystemType, volBindModeWait bool) {
+
+	app := k8stest.FioApplication{
+		Decor:                   decor,
+		VolSizeMb:               4096,
+		OpenEbsEngine:           engine,
+		VolType:                 volType,
+		FsType:                  fstype,
+		Loops:                   1,
+		VolWaitForFirstConsumer: volBindModeWait,
+	}
+
+	loopDevice := e2e_agent.LoopDevice{
+		Size:   10737418240,
+		ImgDir: "/tmp",
+	}
+
+	workerNodes, err := lvm.ListLvmNode(common.NSMayastor())
+	Expect(err).ToNot(HaveOccurred(), "failed to list worker node")
+
+	nodeConfig = lvm.LvmNodesDevicePvVgConfig{
+		VgName:        "lvmvg",
+		NodeDeviceMap: make(map[string]e2e_agent.LoopDevice), // Properly initialize the map
+	}
+	for _, node := range workerNodes {
+		nodeConfig.NodeDeviceMap[node] = loopDevice
+	}
+
+	logf.Log.Info("setup node with loop device, pv and vg", "node config", nodeConfig)
+	err = nodeConfig.ConfigureLvmNodesWithDeviceAndVg()
+	Expect(err).ToNot(HaveOccurred(), "failed to setup node")
+
+	// setup sc parameters
+	app.Lvm = k8stest.LvmOptions{
+		VolGroup:      nodeConfig.VgName,
+		Storage:       "lvm",
+		ThinProvision: common.No,
+	}
+
+	logf.Log.Info("create sc, pvc, fio pod")
+	err = app.DeployApplication()
+	Expect(err).To(BeNil(), "failed to deploy app")
+
+	logf.Log.Info("sleep for 2 min")
+	time.Sleep(2 * time.Minute)
```

Comment:
Shouldn't we wait for the fio pod to complete successfully?

Reply:
yes it should wit for fio pod completion. Removed sleep and added wait for fio pod completion https://github.com/openebs/openebs-e2e/compare/23d1f1e696f8398f09a73c5e048f55e6ddb2f3c7..3951d7a660c87a8e588295d291252b7cac6ef54d